src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -454,19 +464,25 @@ private static boolean skipDoubleBraceInstInit(Details details) {
      * @return true if block has single-line format and is alone on a line.
      */
     private static boolean isBlockAloneOnSingleLine(Details details) {
-        final DetailAST rcurly = details.rcurly;
-        final DetailAST lcurly = details.lcurly;
         DetailAST nextToken = details.nextToken;
-        while (nextToken.getType() == TokenTypes.LITERAL_ELSE) {
+        while (nextToken != null && nextToken.getType() == TokenTypes.LITERAL_ELSE) {
             nextToken = Details.getNextToken(nextToken);
         }
-        if (nextToken.getType() == TokenTypes.DO_WHILE) {
+
+        if (nextToken != null && nextToken.getType() == TokenTypes.DO_WHILE) {
             final DetailAST doWhileSemi = nextToken.getParent().getLastChild();
             nextToken = Details.getNextToken(doWhileSemi);
         }
-        return TokenUtil.areOnSameLine(rcurly, lcurly)
-                && (!TokenUtil.areOnSameLine(rcurly, nextToken)
+        boolean isBlockAloneOnSingleLine = true;
+        if (nextToken != null) {
+            final Optional<DetailAST> rcurly = Optional.ofNullable(details.rcurly);
+            final Optional<DetailAST> lcurly = Optional.ofNullable(details.lcurly);
+            isBlockAloneOnSingleLine = TokenUtil.areOnSameLine(rcurly.orElse(null),
+                    lcurly.orElse(null))
```

Comment:
`lcurly.orElse(null)` such usages do not make sense since it will cause NPE anyway. Point to use optionals is to avoid NPE

Reply:
I had actually use them to avoid `.get()` which according to team city requires `.isPresent()` before, so just to avoid tthat I had used `.orElse()`. Also, some them get never through `null` due to coding constraints.

Reply:
Done